pandas/io/parsers.py

```diff
@@ -235,6 +235,12 @@
     arguments.
 dayfirst : bool, default False
     DD/MM format dates, international and European format.
+cache_dates : boolean, default True
+    If True, use a cache of unique, converted dates to apply the datetime
+    conversion. May produce significant speed-up when parsing duplicate
+    date strings, especially ones with timezone offsets.
+
+    .. versionadded:: 0.25.0
```

Comment:
@anmyachev : How big is the performance increase with your benchmarks? I understand the usefulness of caching, but I'm hesitant to add yet another parameter to this already large signature.

Is there a reason why we couldn't just always have it set to `True` ?

Also, you will need to add this docstring change `doc/source/user_guide/io.rst`.

Reply:
Yes, @gfyoung I see you question, but result of benchmark was not ready until that moment :)
But now:

master|           patch|         ratio| test_name
-|-|-|-
45.4±0.4ms|      6.39±0.06ms|     0.14| io.csv.ReadCSVCachedParseDates.time_read_csv_cached(True)

As you see, we can great boost on parsing dates. The ratio of unique dates to a total number of them, that is, 1/10 is taken from practice, that is reachable on many dataset I think. For example, these ratio on mortgage dataset far less. If I remember correctly - 1/2000(something like that)
> Is there a reason why we couldn't just always have it set to `True` ?

Although the increase in productivity is very, very significant, you should not forget that this incurs additional costs and in some situations, users may simply not have enough memory (then they can simple set `cache_dates` to `False`)

> Also, you will need to add this docstring change `doc/source/user_guide/io.rst`.

I will add the stuff later

Reply:
> Is there a reason why we couldn't just always have it set to `True` ?

Aside from what @anmyachev says about memory note that there's also computing overhead as current cache implementation first finds unique date string indices, then parses said indices, then expands. So if there's not much duplicate dates it would be both memory- and compute-wise costly (though in most data I've seen so far typically unique dates to total dates ratio is pretty small, e.g. in aforementioned Mortgage dataset there are about 400 unique dates out of several milliion entries).